Circuit/Simulation/Simulation.cs

```diff
@@ -86,17 +86,17 @@ public TransientSolution Solution
         /// </summary>
         public Expression SampleRate { get { return 1 / (Solution.TimeStep * oversample); } }
 
-        private IEnumerable<Expression> input = new Expression[] { };
+        private Expression[] input = new Expression[] { };
         /// <summary>
         /// Expressions representing input samples.
         /// </summary>
-        public IEnumerable<Expression> Input { get { return input; } set { input = value.Buffer(); InvalidateProcess(); } }
+        public Expression[] Input { get { return input; } set { input = value; InvalidateProcess(); } }
```

Comment:
How about leaving the public properties as `IEnumerable<Expression>`, but changing `value.Buffer()` to `value.ToArray()` and keeping the (private) member variable an array?

Then the changes to LiveSimulation.xaml.cs aren't necessary, and I think there are also uses of this in Test\Test.cs that will be broken without similar changes.